R/lifetable_indices.R

```diff
@@ -0,0 +1,105 @@
+lt.check <- function(life_table) {
+
+  # check if life_table is of class "mortaar_life_table"
+
+  if (!inherits(life_table,"mortaar_life_table_list")
+      & !inherits(life_table,"mortaar_life_table"))  {
+    paste0(
+      "The dataset does not seem to be a mortAAR life table."
+    ) %>%
+      message
+  } else if (!inherits(life_table,"mortaar_life_table"))  {
+    paste0(
+      "Please select only one life table."
+    ) %>%
+      message
+  }
+}
+
+
+#' Generates indices from a mortAAR life table for use in other functions
+#'
+#' This function bundles a number of indices and vectors from a mortAAR
+#' life table which are needed by other functions in mortAAR.
+#'
+#' @param life_table an object of class mortaar_life_table.
+#'
+#' @return
+#' A list with the following indices and vectors:
+#' \itemize{
+#'   \item \bold{child_i}:   ratio of individuals aged 5--9 to those aged
+#'   10--14 according to Bocquet-Appel and Masset.
+#'   \item \bold{juvenile_i}:   ratio of individuals aged 5--14 to adults
+#'   according to Bocquet-Appel and Masset.
+#'   \item \bold{P(5-19)}:  ratio of individuals aged 5--19 to those aged
+#'   5 or above according to Bocquet-Appel 2002.
+#'   \item \bold{D0_14_D}:   proportion of individuals aged 0--14
+#'   according to McFadden and Oxenham 2017 if infants are represented
+#'   well.
+#'   \item \bold{mortality}:   list of mortalities 5q0, 5q10, 5q15, 45q15.
+#'   \item \bold{repro_indx}: vector of survival rates of individuals
+#'   aged 15--45.
+#'   \item \bold{e0}:   life expectancy at age 0.
+#'}
+#'
+#' @export
```

Comment:
Is it necessary to export this function? Seems to be a helper. Maybe it would be better to split this into many small functions.

Reply:
Yes and no. The indices are so important that I would like to make them visible for the user. But I have split the function to hide the mortality indices and to make the output and the referencing easier.